### PR TITLE
Tap changer table point: ratio is optional

### PR DESCRIPTION
**Pull request by zamarrenolm**
Phase tap changer table points may not define a value for ratio (value 1 is assumed)

### CODE REVIEW

**Comment by geofjamg** [comment]
Is it something clearly defined in CGMES spec?

**Comment by zamarrenolm** [comment]
> Is it something clearly defined in CGMES spec?

Yes it is. In the `TapChangerTablePoint` class, the attribute `ratio` has Range: 0 to 1

**Comment by MioRtia** [comment]
`ratio` is optional for `RatioTapChangerTablePoint` too according to the spec, shouldn't we modify it too ?

**Comment by zamarrenolm** [comment]
> `ratio` is optional for `RatioTapChangerTablePoint` too according to the spec, shouldn't we modify it too ?

Yes, I we should also make it optional. We have not seen it yet in any of the test cases, but as you point out, the spec says it is not required for ratio tap changer table points.

**Comment by zamarrenolm** [comment]
> > `ratio` is optional for `RatioTapChangerTablePoint` too according to the spec, shouldn't we modify it too ?
> 
> Yes, I we should also make it optional. We have not seen it yet in any of the test cases, but as you point out, the spec says it is not required for ratio tap changer table points.

We can keep the PR opened and I will add the required changes.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/PhaseTapChangerConversion.java [hunk 1]
```diff
@@ -130,11 +130,15 @@ private void addStepsFromTable(PhaseTapChangerAdder ptca) {
         String tableId = p.getId("PhaseTapChangerTable");
```
Comment (MioRtia) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/PhaseTapChangerConversion.java]: Maybe you should also test if `tableId` is `null`, it is optional for `PhaseTapChangerTabular` according to the spec.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/RatioTapChangerConversion.java [hunk 2]
```diff
@@ -116,7 +116,7 @@ private void addStepsFromTable(RatioTapChangerAdder rtca) {
         table.sort(byStep);
```
Comment (MioRtia) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/RatioTapChangerConversion.java]: Maybe we should return a `missing` warning here if `table` is empty like in `PhaseTapChangerConversion` ?

Comment (zamarrenolm) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/RatioTapChangerConversion.java]: Added

**Comment by zamarrenolm** [comment]
Optional attributes must be in separate OPTIONAL clauses in the SPARQL query.
